### ISSUE TITLE
[TEST-FAILURE] Java9 - QueryBasicTest.testPredicateDateAttribute

### ISSUE THREAD

**Issue by mdogan** [hazelcast/hazelcast#5980]
`QueryBasicTest.testPredicateDateAttribute()` fails with java 9. 

https://hazelcast-l337.ci.cloudbees.com/view/Quality-Outreach/job/Hazelcast-3.x-OpenJDK9-Quality-Outreach/165/console

```
01:41:52 testPredicateDateAttribute(com.hazelcast.map.query.QueryBasicTest)  Time elapsed: 0.059 sec  <<< ERROR!
01:41:52 java.lang.RuntimeException: Unable to parse date from value: 'Sat Jan 01 11:43:05 EET 2011' ! Valid format are: 'EEE MMM dd HH:mm:ss zzz yyyy'.
01:41:52 	at com.hazelcast.query.impl.DateHelper.throwRuntimeParseException(DateHelper.java:78)
01:41:52 	at com.hazelcast.query.impl.DateHelper.parseDate(DateHelper.java:41)
01:41:52 	at com.hazelcast.query.impl.TypeConverters$DateConverter.convert(TypeConverters.java:113)
01:41:52 	at com.hazelcast.query.impl.predicates.AbstractPredicate.convert(AbstractPredicate.java:70)
01:41:52 	at com.hazelcast.query.impl.predicates.GreaterLessPredicate.apply(GreaterLessPredicate.java:58)
01:41:52 	at com.hazelcast.query.SqlPredicate.apply(SqlPredicate.java:67)
01:41:52 	at com.hazelcast.map.impl.BasicMapContextQuerySupport.queryOnPartition(BasicMapContextQuerySupport.java:93)
01:41:52 	at com.hazelcast.map.impl.operation.QueryPartitionOperation.run(QueryPartitionOperation.java:51)
01:41:52 	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:160)
01:41:52 	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.processOperation(OperationThread.java:180)
01:41:52 	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.process(OperationThread.java:130)
01:41:52 	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.doRun(OperationThread.java:122)
01:41:52 	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.run(OperationThread.java:97)
01:41:52 Caused by: java.text.ParseException: Unparseable date: "Sat Jan 01 11:43:05 EET 2011"
01:41:52 	at java.text.DateFormat.parse(DateFormat.java:366)
01:41:52 	at com.hazelcast.query.impl.DateHelper.parseDate(DateHelper.java:39)
01:41:52 	at com.hazelcast.query.impl.TypeConverters$DateConverter.convert(TypeConverters.java:113)
01:41:52 	at com.hazelcast.query.impl.predicates.AbstractPredicate.convert(AbstractPredicate.java:70)
01:41:52 	at com.hazelcast.query.impl.predicates.GreaterLessPredicate.apply(GreaterLessPredicate.java:58)
01:41:52 	at com.hazelcast.query.SqlPredicate.apply(SqlPredicate.java:67)
01:41:52 	at com.hazelcast.map.impl.BasicMapContextQuerySupport.queryOnPartition(BasicMapContextQuerySupport.java:93)
01:41:52 	at com.hazelcast.map.impl.operation.QueryPartitionOperation.run(QueryPartitionOperation.java:51)
01:41:52 	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:160)
01:41:52 	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.processOperation(OperationThread.java:180)
01:41:52 	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.process(OperationThread.java:130)
01:41:52 	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.doRun(OperationThread.java:122)
01:41:52 	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.run(OperationThread.java:97)
```

**Comment by mdogan** [comment]
I've done a simple test, following is failing with java 9 but passing with 7 and 8.

```java
static final String DATE_FORMAT = "EEE MMM dd HH:mm:ss zzzz yyyy";

public static void main(String[] args) throws Exception {
    DateFormat format = new SimpleDateFormat(DATE_FORMAT, Locale.US);
    Date date = format.parse("Sat Jan 01 11:43:05 EET 2011");
    System.out.println(date);
}
```

If I change date string to `Sat Jan 01 11:43:05 EEST 2011` (EET -> EEST) then it works.

**Comment by hasancelik** [comment]
related issue:
#5800

**Comment by tombujok** [comment]
I've retested it on:
```
$ java -version
java version "1.9.0-ea"
Java(TM) SE Runtime Environment (build 1.9.0-ea-b78)
Java HotSpot(TM) 64-Bit Server VM (build 1.9.0-ea-b78, mixed mode)
```

I could not reproduce it. The following sample code works fine.
```
    @Test
    public void testParse() throws java.lang.Exception {
        DateFormat format = new SimpleDateFormat(DATE_FORMAT, Locale.US);
        Date date1 = format.parse("Sat Jan 01 11:43:05 EET 2011");
        Date date2 = format.parse("Sat Jan 01 11:43:05 EEST 2011");
        System.out.println(date1);
        System.out.println(date2);
    }
```

The output is (keep in mind that EET and EEST are two different time zones): 
```
Sat Jan 01 10:43:05 CET 2011
Sat Jan 01 09:43:05 CET 2011
```

**Comment by tombujok** [comment]
Now fully understand it. Jenkins is running the following Java version (1.9.0-ea-b76).
```
01:22:09 + java -version
01:22:09 java version "1.9.0-ea"
01:22:09 Java(TM) SE Runtime Environment (build 1.9.0-ea-b76)
01:22:09 Java HotSpot(TM) 64-Bit Server VM (build 1.9.0-ea-b76, mixed mode)
```
It can be verified here:
https://hazelcast-l337.ci.cloudbees.com/view/Quality-Outreach/job/Hazelcast-3.x-OpenJDK9-Quality-Outreach/165/console

I've retested on 1.9.0-ea-b78 and hasn't experience this behavior and the reason is that this bug has been fixed in 1.9.0-ea-b77
Here's the list of bugs fixed in b77:
http://download.java.net/jdk9/changes/jdk9-b77.html?q=download/jdk9/changes/jdk9-b77.html

and here's the bug from that list that probably caused our tests to fail:
http://bugs.java.com/bugdatabase/view_bug.do?bug_id=8130845

IMHO this one and #5800 could be closed.
